### PR TITLE
update python syntax to be compatible with 2.6+ and 3.3+

### PR DESCRIPTION
**Pull request by migueldvb**

### CODE REVIEW

File: bin/mksetup [hunk 1]
```diff
@@ -7,7 +7,7 @@ try:
     eupsdir = sys.argv.pop(0)
     prefix = sys.argv.pop(0)
 except IndexError:
-    print >> sys.stderr, "Usage: mksetup EUPS_DIR EUPS_PATH [SETUP_ALIASES]"
+    sys.stderr.write("Usage: mksetup EUPS_DIR EUPS_PATH [SETUP_ALIASES]\n")
```
Comment (timj) [inline bin/mksetup]: if we are switching to Python 2.6+ syntax wouldn't we import `print_function` and use the `file=sys.stderr` scheme?

Comment (migueldvb) [inline bin/mksetup]: @timj I think that makes sense. Both ways work so it is a matter of personal preference.

Comment (timj) [inline bin/mksetup]: Changing a `print` to a `print` seems clearer to me.

**Comment by timj** [comment]
I will be interested to know whether this will be integrated. As recently as 7 months ago (in commit 5d61e00f508162857d975a1645ddd51021db663c ) there were strong indications that python 2.4 must be supported.

**Comment by PaulPrice** [comment]
I suspect not, as eups is used by other projects so we want to keep support for python 2.4.

**Comment by migueldvb** [comment]
Ok, then these changes will not work with python 2.4

**Comment by timj** [comment]
I wasn't suggesting you close it. I suggested waiting for @RobertLuptonTheGood to decide. He's observing this week so wait until next week. One option is that we make a final release that supports 2.4 and then move to a 2.7/3.3+ scheme.

**Comment by migueldvb** [comment]
@timj that sounds good. I'll keep those changes in my python3 branch if they can be useful

**Comment by mjuric** [comment]
Hey @migueldvb,

Thanks for the patchset -- must've been quite a lot of work! Knowing the codebase, I wouldn't dare trying to do this myself :).

I agree it'd be really nice to move up to 2.6-level of compatibility, if we can. The only big machine I know of where the OS version is below 2.6 is NERSC Edison (though 2.7+ are available there through modules). Also, we've had 2.4 compatibility broken sometimes for long periods of time w/o anyone noticing, so maybe it isn't being used much at also? That said, I don't have great insight into the broader EUPS user base; @RobertLuptonTheGood will know that.

I think there are two decisions here: whether to drop 2.4 support (and perhaps even be as bold as going to 2.7?), and then whether this patch is the way to go to do it. The former is a thumbs up/down from the BDFL (once he's back from shining lights down poor Blanco's throat), but the latter will still require a code review (e.g., I'm not sure that changing `print >>fd` constructs to `fd.write` is the way to go, etc.).

**Comment by migueldvb** [comment]
@mjuric Thank you for the comments about the patches. Most of the syntax changes were done using automated translation to edit the code so it was ok :) In the same way it should be possible to have the same code compatible with both versions of python 2.6+ and 3.3+ , but if support for 2.4 is still required then we can not have 2/3 compatibility.

**Comment by RobertLuptonTheGood** [comment]
If you tag the version just before this merge I'm OK with it.

I'd have been much happier if unnecessary changes such as whitespace fixes hadn't been included.

**Comment by timj** [comment]
Great. Are you unhappy with whitespace fixes in general or with them being in the same commit as "real" changes? I'm in favor of a whitespace clean up in general so long as it's a separate commit.

**Comment by RobertLuptonTheGood** [comment]
I'm worried about commits that make merges harder, and I have some experimental real-DB backed code on a branch.

In general, I don't like to make changes that aren't essential and white space is one -- I don't think it's causing any problems.  But it's probably not worth reverting

					R

**Comment by migueldvb** [comment]
In  Python3 mixing tabs and spaces isn't allowed so I see an `IndentationError` without the whitespace change. I have replaced all tabs with whitespace then some of the changes are not really needed if the tab is in a comment.

**Comment by migueldvb** [comment]
Sorry, I have used `reindent.py` that apart from replacing tabs it also trims excess spaces in the end of lines. I will remove those changes since they are not needed here.

**Comment by RobertLuptonTheGood** [comment]
If you're mangling the whitespace because python 3 is being (foolishly!) picky, it's OK to keep the end-of-line changes.

				R

**Comment by migueldvb** [comment]
I have removed most of the unneeded changes in the last commit so it only replaces tabs by spaces in a few places.  This works fine with python 3.

**Comment by mjuric** [comment]
Merged it (after releasing 1.5.10).

Given the change in minimum required version of Python, I'd propose to immediately tag & release this one as 2.0.0 (@RobertLuptonTheGood, sounds OK?).

**Comment by RobertLuptonTheGood** [comment]
Sounds OK

**Pull request merged**